src/d/semantic/caster.d

```diff
@@ -319,6 +319,21 @@ struct Caster(bool isExplicit, alias bailoutOverride = null) {
 					return CastKind.IntToBool;
 				}
 
+				if (isFloat(bt)) {
+					// cast(float) integral|char
+					// (dmd considers char to float a valid implicit cast)
+
+					if (t == Bool) {
```

Comment:
I understand this is a side effect of the goto case above. A good indicator that the goto case really wasn't the way to go.